pulpcore/pulpcore/app/viewsets/base.py

```diff
@@ -61,13 +61,25 @@ def get_resource(uri, model):
             match = resolve(urlparse(uri).path)
         except Resolver404:
             raise serializers.ValidationError(detail=_('URI not valid: {u}').format(u=uri))
-        pk = match.kwargs['pk']
+        if 'pk' in match.kwargs:
+            kwargs = match.kwargs
+        else:
+            kwargs = {}
+            for key, value in match.kwargs.items():
+                if key.endswith('_pk'):
+                    kwargs["{}__pk".format(key[:-3])] = value
+                else:
+                    kwargs[key] = value
         try:
-            return model.objects.get(pk=pk)
+            return model.objects.get(**kwargs)
         except model.DoesNotExist:
-            raise serializers.ValidationError(detail=_('URI not found: {u}').format(u=uri))
+            raise serializers.ValidationError(detail=_('URI {u} not found for {m}.').format(
```

Comment:
This should probably be a ```NotFound``` Error instead of a ```ValidationError```.

http://www.django-rest-framework.org/api-guide/exceptions/#notfound